deduplicate.go

```diff
@@ -1,31 +0,0 @@
-package opslevel_jq_parser
-
-import (
-	"fmt"
-
-	"github.com/opslevel/opslevel-go/v2024"
-)
-
-func Deduplicated[T any](objects []T, keyFunc func(object T) string) []T {
-	out := make([]T, 0)
-	set := make(map[string]struct{})
-	for _, obj := range objects {
-		key := keyFunc(obj)
-		if _, ok := set[key]; ok {
-			continue
-		}
-		set[key] = struct{}{}
-		out = append(out, obj)
-	}
-	return out
-}
-
-func DeduplicatedTools(objects []opslevel.ToolCreateInput) []opslevel.ToolCreateInput {
```

Comment:
`DeduplicatedTools` was removed and I'm not sure how this will effect downstream repos.
Should there be a `changie` log for this PR?

Reply:
I guess this was not yet used?

Reply:
@davidbloss yes indeed, it is not being used anywhere in our repos and was not present in any releases, so it is safe to delete.